rust/rope/src/spans.rs

```diff
@@ -155,6 +155,12 @@ pub struct SpanIter<'a, T: 'a + Clone + Default> {
     ix: usize,
 }
 
+pub struct SpanBackwardsIter<'a, T: 'a + Clone + Default> {
+    cursor: Cursor<'a, SpansInfo<T>>,
+    ix: usize,
+    done: bool,
+}
+
```

Comment:
Instead of using an iterator, I was also considering implementing `Metric<SpansInfo<T>>` instead, however `Metric::prev` and `Metric::next` does not seem to be suitable for `Spans`, because this has to support multiple `Span`s at a given `offset`.